crates/nu-parser/src/parser.rs

```diff
@@ -2450,6 +2455,10 @@ pub fn unescape_string(bytes: &[u8], span: Span) -> (Vec<u8>, Option<ParseError>
 
     let mut idx = 0;
 
+    if !bytes.contains(&b'\\') {
+        return (bytes.to_vec(), None);
+    }
```

Comment:
That sounds like a nice heuristic!

Hopefully the while loop below is accessible for some compiler optimizations.